virtool/config.py

```diff
@@ -159,6 +203,31 @@ def start_server(ctx, host, port, no_check_db, no_check_files, no_client, no_fet
         port=port
     )
 
+    if use_b2c:
+        if config.port != 9950:
+            logger.fatal("registered endpoints for Azure AD B2C require server to run on port 9950")
+            sys.exit(1)
+        if not b2c_client_id or not b2c_client_secret or not b2c_tenant or not b2c_user_flow:
+            logger.fatal("Required B2C client information not provided. \n--use-b2c option requires the following:"
+                         "\n--b2c-client-id"
+                         "\n--b2c-client-secret"
+                         "\n--b2c-tenant"
+                         "\n--b2c-user-flow\n")
+            sys.exit(1)
+
+        msal_app = msal.ClientApplication(
+            client_id=b2c_client_id,
+            authority=f"https://{b2c_tenant}.b2clogin.com/{b2c_tenant}.onmicrosoft.com/{b2c_user_flow}",
+            client_credential=b2c_client_secret,
+        )
+
+        os.environ["B2C_CLIENT_ID"] = b2c_client_id
+        os.environ["B2C_CLIENT_SECRET"] = b2c_client_secret
+        os.environ["B2C_AUTHORITY"] = f"https://{b2c_tenant}.b2clogin.com/{b2c_tenant}.onmicrosoft.com/{b2c_user_flow}"
+
+        config.use_b2c = True
```

Comment:
`Config` represents values provided to the application at runtime. You shouldn't be mutating it.

I'm not sure what is being accomplished here either...

Reply:
I wanted to add some sort of flag to the config values to indicate when the server is running in "b2c mode". I suppose I could remove this and just check for app["msal"] once I make that change.

Reply:
Isn't that going to come from:
```py
@click.option(
    "--use-b2c",
    help="Use OIDC for authentication",
    is_flag=True
)
```

Reply:
Yes, but I'm saving it in config because I reference it later to check that we are in fact supporting b2c login to avoid "force_login" in `http.auth.index_handler()`. Isn't it also a value provided to the application at runtime?